fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogStore.kt

```diff
@@ -0,0 +1,284 @@
+    val toLanguages: List<Language>? = null,
+    val translatedPageTitle: String? = null,
+) : State
+
+/**
+ * Action to dispatch through the `TranslationsStore` to modify `TranslationsDialogState` through the reducer.
+ */
+sealed class TranslationsDialogAction : Action {
+    /**
+     * Invoked when the [TranslationsDialogStore] is added to the fragment.
+     */
+    object InitTranslationsDialog : TranslationsDialogAction()
+
+    /**
+     * When FetchSupportedLanguages is dispatched, an [TranslationsAction.OperationRequestedAction]
+     * will be dispatched to the [BrowserStore]
+     */
+    object FetchSupportedLanguages : TranslationsDialogAction()
+
+    /**
+     * Invoked when the user wants to translate a website.
+     */
+    object TranslateAction : TranslationsDialogAction()
+
+    /**
+     * Invoked when the user wants to restore the website to its original pre-translated content.
+     */
+    object RestoreTranslation : TranslationsDialogAction()
+
+    /**
+     * Invoked when a translation error occurs during the translation process.
+     */
+    data class UpdateTranslationError(
+        val translationError: TranslationError? = null,
+    ) : TranslationsDialogAction()
+
+    /**
+     *  Updates translate from languages list.
+     */
+    data class UpdateTranslateFromLanguages(
+        val translateFromLanguages: List<Language>,
+    ) : TranslationsDialogAction()
+
+    /**
+     *  Updates translate to languages list.
+     */
+    data class UpdateTranslateToLanguages(
+        val translateToLanguages: List<Language>,
+    ) : TranslationsDialogAction()
+
+    /**
+     *  Updates to the current selected language from the "translateFromLanguages" list.
+     */
+    data class UpdateFromSelectedLanguage(
+        val language: Language,
+    ) : TranslationsDialogAction()
+
+    /**
+     *  Updates to the current selected language from the "translateToLanguages" list.
+     */
+    data class UpdateToSelectedLanguage(
+        val language: Language,
+    ) : TranslationsDialogAction()
+
+    /**
+     * Dismiss the translation dialog fragment.
+     */
+    data class DismissDialog(
+        val dismissDialogState: DismissDialogState,
+    ) : TranslationsDialogAction()
+
+    /**
+     * Updates the dialog content to translation in progress status.
+     */
+    data class UpdateTranslationInProgress(
+        val inProgress: Boolean,
+    ) : TranslationsDialogAction()
+
+    /**
+     * Updates the dialog content to translated status.
+     */
+    data class UpdateTranslated(
+        val isTranslated: Boolean,
+    ) : TranslationsDialogAction()
+
+    /**
+     * Updates the dialog title if the page was translated.
+     */
+    data class UpdateTranslatedPageTitle(val title: String) : TranslationsDialogAction()
+}
+
+/**
+ * Positive button type from the translation bottom sheet.
+ */
+sealed class PositiveButtonType {
+    /**
+     * The download indicator will appear.
```

Comment:
```suggestion
     * The download indicator will appear.
```
Did you mean translating indicator?

Reply:
Done